random-menu-generator.rb

```diff
@@ -0,0 +1,784 @@
+# response if no key letters found.
+def request_user_confirmation
+  user_confirmation = get_user_input(true)
+
+  if user_confirmation.include? "n"
+    return false
+  elsif user_confirmation.include? "y"
+    return true
+  else
+    possible_responses = [true, false]
+    random_choice = rand(0...possible_responses.length)
+    return possible_responses[random_choice]
+  end
+end
+
+
+
+
+# checks if a user input string contains any keywords associated with various
+# features and then calls the next step for whichever feature.
+def verify_feeling(user_feeling)
+  creative = %w{creat add update more mine my own}
+  hungry = %w{hung angry peckish eat food dinner lunch breakfast dessert snack dish}
+  ingredients = %w{gred cook reci assem proc instr}
+  world_domination = %w{lazy bored take world hotdog elephant tacos}
+
+  creative.each do |keyword| # create is first, because 'eat' is part of it
+    if user_feeling.include? keyword
+      return create_user_ideas
+    end
+  end
+
+  hungry.each do |keyword| # eat is a trigger here, but create has precedence
+    if user_feeling.include? keyword
+      $individual = false
+      return retrieve_ideas("dishes")
+    end
+  end
+
+  ingredients.each do |keyword|
+    if user_feeling.include? keyword
+      $individual = true
+      return retrieve_ideas("hungry")
+    end
+  end
+
+  world_domination.each do |keyword|
+    if user_feeling.include? keyword
+      $individual = true
+      return retrieve_ideas("bored")
+    end
+  end
+
+
+  speak("You said #{user_feeling}. I'm sorry. I don't understand that emotion yet.")
+  speak("Until I learn your emotion, I will interpret that as hungry / angry.") # !W? this is sort of a lie b/c keyboard mismatch w/ name
+  $individual = true
+  return retrieve_ideas("dishes")
+end
+
+
+
+
+# after stripping spaces and commas out a provided string, verifies whether
+# said string is qualified to be turned into a number. returns it as an integer
+# if so. gives the user a hint and calls itself again if not.
+def verify_number(user_number)
+  if user_number.include? " "
+    user_number.strip!
+  end
+  while user_number.include? ","
+    user_number.slice!(",")
+  end
+
+  if (user_number > "/") && (user_number < ":")
+    return user_number.to_i
+  else
+    speak("You said #{user_number}. That doesn't look like a number to me.")
+    speak("Please provide your number in numeric format (eg, 12, -4,003.13).")
+    return verify_number(get_user_input(false))
+  end
+end
+
+
+
+
+# returns the length of the smallest of three arrays within an (outer) array.
+# used to assign a max value for the select random ideas function, so it will
+# not try to add empty items to a set.
+def assign_max_value(array_of_ideas)
+  a = array_of_ideas[0].length
+  b = array_of_ideas[1].length
+  c = array_of_ideas[2].length
+
+  if (a <= b) && (a <= c)
+    return a
+  elsif (b <= c) && (b <= a)
+    return b
+  else # (c <= a) && (c <= b)
+    return c
+  end
+end
```

Comment:
You could use [`max`](http://ruby-doc.org/core-2.2.2/Enumerable.html#method-i-max) here:

```ruby
idea_lengths = array_of_ideas.map { |ideas| ideas.length }
return idea_lengths.max
```